crates/router/src/db/business_profile.rs

```diff
@@ -114,38 +114,89 @@ impl BusinessProfileInterface for Store {
 impl BusinessProfileInterface for MockDb {
     async fn insert_business_profile(
         &self,
-        _business_profile: business_profile::BusinessProfileNew,
+        business_profile: business_profile::BusinessProfileNew,
     ) -> CustomResult<business_profile::BusinessProfile, errors::StorageError> {
-        Err(errors::StorageError::MockDbError)?
+        let business_profile_insert = business_profile::BusinessProfile::from(business_profile);
+        self.business_profiles
+            .lock()
+            .await
+            .push(business_profile_insert.clone());
+        Ok(business_profile_insert)
     }
 
     async fn find_business_profile_by_profile_id(
         &self,
-        _profile_id: &str,
+        profile_id: &str,
     ) -> CustomResult<business_profile::BusinessProfile, errors::StorageError> {
-        Err(errors::StorageError::MockDbError)?
+        self.business_profiles
+            .lock()
+            .await
+            .iter()
+            .find(|business_profile| business_profile.profile_id == profile_id)
+            .ok_or(
+                errors::StorageError::ValueNotFound(
+                    "No business profile found for profile_id = {profile_id}".to_string(),
```

Comment:
Use `format!` marco here too.
```suggestion
                    format!("No business profile found for profile_id = {}", profile_id),
```